### PR TITLE
linux: Enable building for Linux platforms with Crashpad.

### LINKED ISSUE #232: Crashpad support for Linux
Are there plans to support crashpad on Linux?

### PR DESCRIPTION
**Pull request by jblazquez**
This change fixes a compile error when MemorySanitizer is enabled and also allows sentry-native to be built targeting the Linux platform with the Crashpad backend. Please let me know if you actually don't want to support this backend on Linux I can close this PR.

This depends on the changes here being merged: https://github.com/getsentry/crashpad/pull/15

Fixes #232

### CODE REVIEW

File: src/sentry_unix_pageallocator.c [hunk 1]
```diff
@@ -60,7 +60,7 @@ get_pages(size_t num_pages)
 
 #if defined(__has_feature)
 #    if __has_feature(memory_sanitizer)
-    __msan_unpoison(a, page_size_ * num_pages);
+    __msan_unpoison(rv, g_alloc->page_size * num_pages);
```
Comment (jblazquez) [inline src/sentry_unix_pageallocator.c:63]: I _think_ this was the intent of this code. The variables just seemed to have been renamed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -79,9 +79,6 @@ else()
 	message(FATAL_ERROR "SENTRY_BACKEND must be one of 'crashpad', 'inproc', 'breakpad' or 'none'")
 endif()
 
-if(SENTRY_BACKEND_CRASHPAD AND NOT (APPLE OR WIN32))
```
Comment (Swatinem) [inline CMakeLists.txt:82]: I would rather flag this with `AND ANDROID`, since we don’t test/support that backend on android yet.

**Comment by Swatinem** [comment]
With your crashpad PR merged, you can also bump the submodule ;-)

File: src/sentry_unix_pageallocator.c (hunk 1 above)
Comment (Mixaill) [reply src/sentry_unix_pageallocator.c:63]: It was changed by @blinkov in 
https://github.com/getsentry/sentry-native/commit/5dd159f64094c412b99be654d95c3cbabd769090

File: CMakeLists.txt (hunk 2 above)
Comment (jblazquez) [reply CMakeLists.txt:82]: Good callout.

**Comment by jblazquez** [comment]
@Swatinem, updated the PR with your feedback.

**Comment by jan-auer** [comment]
Fixes #232

**Comment by jan-auer** [comment]
@Swatinem before we merge this in, could you please check if the HTTP implementation in Crashpad works now? The last time I checked, Crashpad was not able to make HTTPS requests and generally had troubles with getting requests out.

**Comment by Swatinem** [comment]
Adding at least a smoketest for crashpad here: #310 
Although that runs on http right now, without TLS.

**Comment by jblazquez** [comment]
Thanks Arpad, I updated the tests script.

**Comment by jblazquez** [comment]
Looks like CI broke after enabling Crashpad tests on Linux, something related to `pytest`:

https://dev.azure.com/sentry-builds/sentry-native/_build/results?buildId=2534&view=logs&j=ad746dd1-389b-5d3a-67ba-32f8fe0c787d&t=ac971eda-1633-5f8a-d283-de94f79a855c&l=3552

```
NameError: name 'pytest' is not defined
```

I'm gonna have to get familiar with the CI setup to understand what's going on, but if you have any ideas please let me know.

**Comment by Swatinem** [comment]
Ah, that is just my oversight.
The real error is this one:

```
In file included from /home/vsts/work/1/s/include/sentry.h:81:
/home/vsts/work/1/s/external/crashpad/compat/linux/signal.h:18:2: error: #include_next is a language extension [-Werror,-Wgnu-include-next]
#include_next <signal.h>
 ^
1 error generated.
```

I remember we had one of these already. Also I’m surprised that all the compat includes are leaking into sentry.

**Comment by jan-auer** [comment]
Crashpad integration tests have moved to https://github.com/getsentry/sentry-native/pull/311

**Comment by Swatinem** [comment]
This was done as part of #320 now

**Pull request closed without merging**